[docs] Row and cell editing new examples for server-side persistence (#3124)
mui-org/material-ui-x: docs/src/pages/components/data-grid/editing/editing.md

{{"demo": "pages/components/data-grid/editing/BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}

> ### Server-side persistence

Reviewer: It should not be the first section. The user has just opened the editing page to learn how it works. I think it could be below "Server-side validation".

<img width="150" src="https://user-images.githubusercontent.com/42154031/143038577-f3104f83-54e8-4cd5-9075-b1a735cfa6a7.png" alt="image" style="max-width: 100%;">

Author: Done

> ### Service-side persistence [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)

Reviewer: ```suggestion
### Server-side persistence [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)
```

> If you want to send the edited data to your server, you can use the `onCellEditCommit` which is fired just before committing the new row value to the grid.

Reviewer: ```suggestion
If you want to send the updated data to your server, you can use the `onCellEditCommit` which is fired just before committing the new cell value to the grid.
```

> You can then decide if you want to send the whole row or manually check the modified fields and only send those.

Reviewer: ```suggestion
You can then decide if you want to send the whole row or only the modified fields.
```

{{"demo": "pages/components/data-grid/editing/BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}

### Service-side persistence [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)

> If you want to send the edited data to your server, you can use the `onRowEditCommit` which is fired just before committing the new row value to the grid.

Reviewer: ```suggestion
If you want to send the updated data to the server, you can use the `onRowEditCommit` prop which is fired just before committing the new row value to the grid.
To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.
```

Author: If I stop using the `apiRef` in the demo, then we should probably add a "If you are using the pro plan" here.

{{"demo": "pages/components/data-grid/editing/BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}

### Server-side persistence [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)

If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.

To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.

> You can then decide if you want to send the whole row or only the modified fields.

Reviewer: ```suggestion
You can then decide if you want to send the whole row or only the modified fields, by checking it against the previous row values.
```

The same [approach](/components/data-grid/editing/#saving-nested-structures) from the cell editing mode can be used here.
Note that the `valueSetter` will be called for each field.

### Server-side persistence [<span class="pro"></span>](https://mui.com/store/items/material-ui-pro/)

If you want to send the updated data to your server, you can use the `onRowEditCommit` which is fired just before committing the new cell value to the grid.

To access the new values for the row, use `apiRef.current.getEditRowsModel` to get all rows in edit mode, then use the id provided to get only the values for the row that was committed.

> You can then decide if you want to send the whole row or only the modified fields, by checking it against the previous row values.

Reviewer: ```suggestion
You can then decide if you want to send the whole row or only the modified fields, by checking them against the previous row values.
```
